### PR TITLE
libsubprocess: remove prefix on server setup

### LINKED ISSUE #4176: libsubprocess: make `flux_rexec()` topic configurable OR remove server topic configuration
In `libsubprocess` the `flux_rexec()` function sends remote execs to the `broker.rexec` topic.

The `flux_server_start()` function takes a prefix to determine what the server target `<prefix>.rexec` will be called.  The current flux broker sets the prefix to `broker`, so everything works.

This is inconsistent, presumably one of these should be adjusted.  The removal of the `prefix` seems the easier path.  I cannot find a reason why the `prefix` is supported, it may have been a legacy reason that is now forgotten.

### PR DESCRIPTION
**Pull request by chu11**
Problem: The flux_subprocess_server_start() function takes
a prefix for setting up server targets.  However the flux_rexec()
function is not configurable on what target should be used to
launch a remote subprocess.  This inconsistency is likely due
to a forgotten design decision / hope from long ago.

Solution: Remove the prefix for server setup and always internally
assume the prefix is "broker".

Fixes #4176

### CODE REVIEW

**Comment by chu11** [comment]
an aside, this inconsistency has been annoying me, so just decided to fix it.  can revisit in the future if we ever need to add this support for real (would probably require change to both `flux_subprocess_server_start()` and `flux_rexec()`).

**Comment by chu11** [comment]
> I have a branch from awhile ago that uses this functionality.

Ohhh I didn't realize that you were actually using it.  How are you handling config of the target via `flux_rexec()`?

**Comment by grondo** [comment]
I had forgotten too until I remembered. I added a `flux_cmd_setopt` option, e.g.

```c
    flux_cmd_setopt (cmd, "service", "shell-123456.rexec");
```

**Comment by chu11** [comment]
> I had forgotten too until I remembered. I added a flux_cmd_setopt option, e.g.

ahh ok.  Well, I'll leave it to you.  Do you think you'll need it with reasonable probability in near future?  We can drop this PR of course.

**Comment by grondo** [comment]
I'm not sure there's a use case for it. We were thinking it could maybe be useful for co-locating services like debuggers, but I'm not sure that's needed since we also have "alloc-bypass" capability. We can merge this PR if you'd like and I'll just add a revert to my stack of commits. If we never do add the exec functionality to the shell, it might be nice to have this cleanup in place anyway...

**Comment by chu11** [comment]
> it might be nice to have this cleanup in place anyway...

Yeah, and if we really need this functionality, I figure we need to think a bit on how to do this with `flux_rexec()`.  I sorta didn't want to think about it right now, which is why I went in this direction vs "allow alternate targets with rexec" path.

**Review by grondo — APPROVED** [review]
Seems good to me. We can easily add the prefix back if we ever need it.

**Comment by grondo** [comment]
>  I figure we need to think a bit on how to do this with flux_rexec(). I sorta didn't want to think about it right now, which is why I went in this direction vs "allow alternate targets with rexec" path.


Yeah, and I wouldn't want to add the flux_rexec() changes without a use case, along with all the extra required tests, etc. Let's just obliterate this for now and resurrect it if necessary in the future.

**Pull request merged**
